src/main/java/UserRepository.java

```
@@ -1,3 +1,71 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * Description: This class manages user data storage and retrieval.
+ *
+ * Intent: The UserRepository class handles the storage and retrieval of user objects.
+ * provides methods to add, retrieve, update, and remove users from the repository
+ * using their unique identifier or username. It also ensures
+ * uniqueness of usernames to prevent duplicates.
+ */
 public class UserRepository {
-    // TODO implement class
+
+    private List<User> users;
+
+    public UserRepository() {
+        this.users = new ArrayList<>();
+    }
+
+    public void addUser(User user) {
+        if (isUnique(user)) {
+            users.add(user);
+        } else {
+            throw new RuntimeException("User is already in repository.");
```

Do we really want to throw an exception? Maybe just return false?